extensions/shared/upgrade-nudge/index.jsx

```diff
@@ -73,8 +74,9 @@ export default compose( [
 			upgradeUrl,
 		};
 	} ),
-	withDispatch( ( dispatch, { upgradeUrl } ) => ( {
+	withDispatch( ( dispatch, { plan: plan_slug, upgradeUrl } ) => ( {
 		autosaveAndRedirectToUpgrade: async () => {
+			analytics.tracks.recordEvent( 'jetpack_editor_block_upgrade_click', { plan_slug } );
```

Comment:
Let's use {plan: plan_slug} and use the block name in the object prop: {plan: plan_slug, source: block_name }.

Reply:
Good plan, now we see:

> Record event `"jetpack_editor_block_upgrade_click"` called with props `{"plan":"jetpack_premium","source":"jetpack/simple-payments"}`

Reply:
Is `source` clear for usage? I seem to vaguely recall it might be used as an interal prop by tracks...

Reply:
737880da8